tests/MongoDB.Driver.Core.Tests/Core/Servers/ServerMonitorTests.cs

```
@@ -470,7 +565,7 @@ private CommandResponseMessage CreateHeartbeatCommandResponseMessage(bool moreTo
                     new BsonDocument
                     {
                         { "processId", ObjectId.Parse("5ee3f0963109d4fe5e71dd28") },
-                        { "counter", 0 }
+                        { "counter", new BsonInt64(0) }
```

This was a bug as the `Topology` get method from `HelloResult` expects this to be a 64-bit int and thus silently fails omitting the topology version even though it was provided in the response. This caused my tests to not behave properly even when I knew the server supported streaming based on the responses.

---

nice find